### DISCUSSION TITLE
websend problem with v11.1.0

### DISCUSSION THREAD

**Discussion by pearson** [arendst/Tasmota#15413] [Q&A] [answered]
I'm attempting to use WebSend between two devices flashed with Tasmota 11.1.0. WebSend does not work while a command sent through a web browser does work.  

* `WebSend [10.0.1.4] POWER1 TOGGLE` from the console of device A to device B, device B's web debug log will only show `HTP: Command` with no follow-on lines
* Using `WebSend [10.0.1.4] /cm?cmnd=POWER1%20TOGGLE` or `WebQuery http://10.0.1.4/cm?cmnd=POWER1%20TOGGLE GET` has the same result: only the `HTP: Command` shows in the debug log
* Using a web browser with `http://10.0.1.4/cm?cmnd=POWER1%20TOGGLE`, the command is executed correctly, and `HTP: Command` has its expected follow-on log lines

Am I missing something obvious here?

Thanks!

**EDIT:** The short answer is yes, I was missing something obvious. I had a web password set on the device, but wasn't sending the username/password with the `WebSend` command.

**Comment by sfromis** [comment, marked as answer]
In a small test (with 11.1.0.1), I saw no such problem. Tried both WebSend variants, and also WebQuery.

If you want to try to see what Tasmota gets, try again after activating `Weblog 3` on the target device.

**Reply by pearson** [reply]
@sfromis , I've had weblog set to 3 and to 4.

With weblog set to 4 on both devices, the sender log will show:
```
00:44:44.796 CMD: websend [10.0.1.4] POWER TOGGLE
00:44:44.801 CMD: Grp 0, Cmd 'WEBSEND', Idx 1, Len 24, Pld -99, Data '[10.0.1.4] POWER TOGGLE'
00:44:44.848 MQT: stat/tasmota_4EF150/RESULT = {"WebSend":"Done"}
```
and the receiver log will show:
```
00:44:45.209 HTP: Command
```

However, if I do a command through the web browser, the receiver log will show:
```
00:46:08.549 HTP: Command
00:46:08.554 SRC: WebCommand from 10.0.1.2
00:46:08.557 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 6, Pld 2, Data 'TOGGLE'
00:46:08.563 MQT: stat/tasmota_4EF16C/RESULT = {"POWER":"ON"}
00:46:08.566 MQT: stat/tasmota_4EF16C/POWER = ON
00:46:08.835 CFG: Saved to flash at FA, Count 26, Bytes 4096
```

I'm looking to set up WireShark to intercept the packets to see what's different here.

**Reply by barbudor** [reply]
Check SetOption128 on the target device
If not set to 0, set it to 0
I'm not sure WedSend/WebQuery set the referer

**Reply by sfromis** [reply]
I actually did a test with `SetOption128 0`, but blocking referer-less requests lead to rather different symptoms, with a very clear message in the console, `Referer '' denied. Use 'SO128 1' for HTTP API commands. 'Webpassword' is recommended`, and also `{"WebSend":"Connect failed"}`.

Are you using a standard precompiled build (which)? Or have you done your own build with a different set of `#define` / `#undef` ?

**Reply by pearson** [reply]
@barbudor , I tried `SetOption 128 0` and I get the same result as @sfromis .

However, this led to the source of the problem: I had set a web password. Using `WebPassword 0` to disable the web password solves the problem.

I'd obviously missed the fact I needed to use the web password for commands. 

I would suggest that the logging be expanded here so that it displays that the command is failing due to a missing (or incorrect) password.

Thanks for the help!
